### PR TITLE
gha: Add coverage for policy secret sync

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to make sure that we have the coverage for policy secret
sync enabled and disabled.

Relates: #35513

### CODE REVIEW

File: .github/workflows/conformance-kind-proxy-embedded.yaml [hunk 1]
```diff
@@ -62,6 +68,7 @@ jobs:
             --helm-set-string=kubeProxyReplacement=true \
             --helm-set=loadBalancer.l7.backend=envoy \
             --helm-set=tls.secretsBackend=k8s \
+            --helm-set=tls.secretSync.enabled=${{ matrix.enable-policy-secret-sync }} \
```
Comment (aanm) [inline .github/workflows/conformance-kind-proxy-embedded.yaml]: Do we need to test it this way? I assume we don't need to run all of the connectivity tests to test with secret sync enabled and vice-versa, no?

**Review by aanm — CHANGES_REQUESTED** [review]

File: .github/workflows/conformance-kind-proxy-embedded.yaml (hunk 1 above)
Comment (sayboras) [reply .github/workflows/conformance-kind-proxy-embedded.yaml]: are you suggesting that we can run relevant tests after flag update in the same job?

Comment (aanm) [reply .github/workflows/conformance-kind-proxy-embedded.yaml]: Correct.

File: .github/workflows/conformance-kind-proxy-embedded.yaml [hunk 2]
```diff
@@ -118,6 +118,24 @@ jobs:
...
+          cilium status --wait --wait-duration=10m
+          kubectl -n kube-system get pods
+
+      - name: Run L7 related connectivity test
+        run: |
+          cilium connectivity test --test="l7|sni|check-log-errors" \
+            --junit-file "cilium-junits/${{ env.job_name }}-secret-sync.xml" \
+            --junit-property github_job_step="Run connectivity test with secret sync disabled"
+
       - name: Features tested
```
Comment (aanm) [inline .github/workflows/conformance-kind-proxy-embedded.yaml:145]: Please copy this step and put it on line 120. Then change line 143 to be `json-filename: "${{ env.job_name }} - with secret sync"`.

This way we are capturing the coverage for both connectivity test cases.

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Review by brlbil — APPROVED** [review]

File: .github/workflows/conformance-kind-proxy-embedded.yaml [hunk 3]
```diff
@@ -124,6 +124,30 @@ jobs:
...
+
+          kubectl -n kube-system rollout restart ds/cilium deployment/cilium-operator
+          cilium status --wait --wait-duration=10m
+          kubectl -n kube-system get pods
+
+      - name: Run L7 related connectivity test
+        run: |
+          cilium connectivity test --test="l7|sni|check-log-errors" \
+            --junit-file "cilium-junits/${{ env.job_name }}-without-secret-sync.xml" \
+            --junit-property github_job_step="Run connectivity test with secret sync disabled"
```
Comment (joestringer) [inline .github/workflows/conformance-kind-proxy-embedded.yaml:143]: @brlbil @cilium/ci-structure I would expect CI structure review to catch that this workflow was not reusing `${{ steps.vars.outputs.connectivity_test_defaults }}` and hence introduced a few CI bugs here such as failing to capture sysdump on failure.

Fix is here: https://github.com/cilium/cilium/pull/41036

**Pull request merged**